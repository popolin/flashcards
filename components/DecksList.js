import React, { Component } from 'react';
import { Button, View } from 'react-native';
import { List, ListItem } from 'react-native-elements'
import { connect } from 'react-redux'
import * as API from '../utils/api'
import { fetchDecks } from "../actions/index";

export class DecksList extends Component {

    componentDidMount() {
        API.fetchDecks().then(decks => {
            this.props.fetchDecks(decks)
        })
    }

    render() {
        const { navigation, decks } = this.props
        return (
            <List>
                {Object.keys(decks).map(key => (
                    <ListItem key={key}
                        onPress={() => navigation.navigate('DeckDetail', {keyDeck: key})}
                        title={decks[key].title}
                        subtitle={`${decks[key].questions.length} cards`}
                    />
                ))}
            </List>
        )
    }

}

const mapStateToProps = (state) => ({decks: state})
const mapDispatchToProps = { fetchDecks }
export default connect(mapStateToProps, mapDispatchToProps)(DecksList)