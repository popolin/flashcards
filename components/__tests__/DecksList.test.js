import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import MockAsyncStorage from 'mock-async-storage'
import {AsyncStorage as storage} from 'react-native'

import DecksList from '../DecksList';

const decks = {
    React: {
        title: 'React',
        questions: [
            {
                question: 'What is React?',
                answer: 'A library for managing user interfaces'
            },
            {
                question: 'Where do you make Ajax requests in React?',
                answer: 'The componentDidMount lifecycle event'
            }
        ]
    }
}

const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const store = mockStore({})
const props = {
}

const mock = () => {
    const mockImpl = new MockAsyncStorage()
    jest.mock('AsyncStorage', () => mockImpl)
}


describe('Deck List', () => {

    it('renders without crashing', async () => {
        mock()

        await storage.setItem('flashcards', JSON.stringify(decks))

        store.dispatch = jest.fn();

        const rendered = renderer.create(<DecksList {...props} store={store} />).toJSON();
        expect(rendered).toMatchSnapshot();
    });

})
