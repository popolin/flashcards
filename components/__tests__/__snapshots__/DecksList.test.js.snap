// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck List renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
/>
`;
